SingularityUI/app/components/tasks/Columns.jsx

```diff
@@ -0,0 +1,297 @@
+    }
+    cellRender={
+      (cellData) => (
+        <span>{cellData} MB</span>
+      )
+    }
+    sortable={true}
+  />
+);
+
+export const ActiveActions = (onTaskKill) => {
+  return <Column
+    label=""
+    id="actions"
+    key="actions"
+    className="actions-column"
+    cellData={
+      (rowData) => rowData
+    }
+    cellRender={(cellData) => {
+        return (
+          <div className="hidden-xs">
+            <a data-action="remove" onClick={() => onTaskKill(cellData.taskId.id)}><Glyphicon iconClass="remove" /></a>
+            <JSONButton className="inline" object={cellData}>
+              {'{ }'}
+            </JSONButton>
+          </div>
+        );
+      }
+    }
+  />
+};
+
+export const NextRun = (
+  <Column
+    label="Next Run"
+    id="nextRun"
+    key="nextRun"
+    cellData={
+      (rowData) => rowData.pendingTask.pendingTaskId.nextRunAt
+    }
+    cellRender={(cellData) => {
+        let label = <span className="label label-default">SCHEDULED</span>;
+        if (Utils.timestampWithinSeconds(cellData, config.pendingWithinSeconds)) {
+          label = <span className="label label-info">PENDING</span>
+        } else if (cellData < Date.now() - config.pendingWithinSeconds * 1000) {
+          label = <span className="label label-danger">OVERDUE</span>
+        }
+        return (
+          <div>
+            {Utils.timeStampFromNow(cellData)} {label}
+          </div>
+        );
+      }
+    }
+    sortable={true}
+  />
+);
+
+export const PendingType = (
+  <Column
+    label="Pending Type"
+    id="pendingType"
+    key="pendingType"
+    cellData={
+      (rowData) => rowData.pendingTask.pendingTaskId.pendingType
+    }
+    cellRender={(cellData) => {
+        return (
+          <div>
+            {Utils.humanizeText(cellData)}
+          </div>
+        );
+      }
+    }
+    sortable={true}
+  />
+);
+
+export const DeployId = (
+  <Column
+    label="Deploy ID"
+    id="deployId"
+    key="deployId"
+    cellData={
+      (rowData) => rowData.pendingTask.pendingTaskId
+    }
+    cellRender={(cellData) => {
+        return (
+          <a href={`${config.appRoot}/request/${cellData.requestId}/deploy/${cellData.deployId}`}>{cellData.deployId}</a>
+        );
+      }
+    }
+    sortable={true}
+  />
+);
+
+export const ScheduledActions = (onRunNow) => {
+  return <Column
+    label=""
+    id="actions"
+    key="actions"
+    className="actions-column"
+    cellData={
+      (rowData) => rowData
+    }
+    cellRender={(cellData) => {
+        return (
+          <div className="hidden-xs">
+            <a onClick={() => onRunNow(cellData.pendingTask.pendingTaskId.requestId)}><Glyphicon iconClass="flash" /></a>
+            <JSONButton className="inline" object={cellData}>
+              {'{ }'}
+            </JSONButton>
+          </div>
+        );
+      }
+    }
+  />;
+};
+
+export const ScheduledTaskId = (
```

Comment:
Yeah, this was the cleanest way I could find to do this. Even the ones that are functions aren't used as react components, just as functions that generate elements. The alternative is to extend the Column class each time you want to save a column, and that's a little too much boilerplate for this kind of use.